src/main/java/com/example/java/maven/game2048/Game.java

```diff
@@ -7,9 +7,74 @@
     public Game(int winningNumber, Board board) {
         this.winningNumber = winningNumber;
         this.board = board;
+        board.placeTwoInitialElementsOnBoard();
+        MessagePrinter.printInstructionsBeforeGame();
     }
 
+
     public void start() {
+        while (!board.checkIfGameOver()) {
```

Comment:
it is not bad, but you can see here: `board.checkIfGameOver()` that on board you are checking something specific to game. `Board` should NOT know anything about game - `Board` class should have methods only related to `Board`.
So:
Introduce method `isNotOver()` in `Game` class - this method will use some methods from boards, for example you can add method: `numberOfFieldsWithZeroValues()` to `Board` class and use it in `isNotOver()`

Reply:
Done